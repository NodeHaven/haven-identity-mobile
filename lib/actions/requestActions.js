// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  HANDLE_URL,
  HANDLE_MESSAGE,
  SELECT_REQUEST,
  SELECT_ACCOUNT_FOR_REQUEST,
  SAVE_REQUEST,
  CLEAR_REQUEST,
  AUTHORIZE_REQUEST,
  CANCEL_REQUEST,
  CALLBACK,
  AUTHORIZE_ACCOUNT
} from 'uPortMobile/lib/constants/RequestActionTypes'

export function handleURL (url, options = {}) {
  return {
    type: HANDLE_URL,
    url,
    ...{postback: false, popup: true, clientId: null},
    ...options
  }
}

export function handleMessage (message, options = {}) {
  return {
    type: HANDLE_MESSAGE,
    message,
    ...options
  }
}

export function selectRequest (activityId) {
  return {
    type: SELECT_REQUEST,
    activityId
  }
}

// Use this in a request dialog to select the correct account
export function selectAccountForRequest (activityId, account) {
  return {
    type: SELECT_ACCOUNT_FOR_REQUEST,
    activityId,
    account
  }
}

export function saveRequest (activityId) {
  return {
    type: SAVE_REQUEST,
    activityId
  }
}

export function clearRequest () {
  return {
    type: CLEAR_REQUEST
  }
}

export function authorizeRequest (activityId) {
  return {
    type: AUTHORIZE_REQUEST,
    activityId
  }
}

export function authorizeAccount (activityId, accType) {
  return {
    type: AUTHORIZE_ACCOUNT,
    activityId: activityId,
    accType: accType
  }
}

export function cancelRequest (activityId) {
  return {
    type: CANCEL_REQUEST,
    activityId
  }
}

export function callback (request) {
  return {
    type: CALLBACK,
    request
  }
}
