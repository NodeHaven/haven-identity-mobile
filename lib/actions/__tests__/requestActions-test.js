// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

import {handleURL, handleMessage, authorizeRequest, cancelRequest, selectRequest, selectAccountForRequest, clearRequest, saveRequest } from 'uPortMobile/lib/actions/requestActions.js'

it('creates a HANDLE_URL action with default postback', () => {
  expect(handleURL('ethereum:0x')).toMatchSnapshot()
})

it('creates a HANDLE_MESSAGE action with default postback', () => {
  expect(handleMessage('eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJpc3MiOiIzNHdqc3h3dmR1YW5vN05GQzh1ak5KbkZqYmFjZ1llV0E4bSIsImlhdCI6MTQ4NTMyMTEzMywiY2xhaW1zIjp7Im5hbWUiOiJCb2IifSwiZXhwIjoxNDg1NDA3NTMzfQ.sg1oJ7J_f2pWaX2JwqzA61oWMUK5v0LYVxUp3PvG7Y25CVYWPyQ6UhA7U9d4w3Ny74k7ryMaUz7En5RSL4pyXg')).toMatchSnapshot()
})

it('creates a HANDLE_URL action with postback', () => {
  expect(handleURL('ethereum:0x', {postback: true})).toMatchSnapshot()
})

it('creates a HANDLE_URL action without postback', () => {
  expect(handleURL('ethereum:0x', {postback: false})).toMatchSnapshot()
})

it('creates a HANDLE_URL action with postback, no popup and client id', () => {
  expect(handleURL('ethereum:0x', {postback: true, popup: false, clientId: '0x010203'})).toMatchSnapshot()
})

it('creates a AUTHORIZE_REQUEST action', () => {
  expect(authorizeRequest(123)).toMatchSnapshot()
})

it('creates a CANCEL_REQUEST action', () => {
  expect(cancelRequest(123)).toMatchSnapshot()
})

it('creates a SELECT_REQUEST action', () => {
  expect(selectRequest(123)).toMatchSnapshot()
})

it('creates a SELECT_REQUEST_FOR_ACCOUNT action', () => {
  expect(selectAccountForRequest(123, '0x234')).toMatchSnapshot()
})

it('creates a CLEAR_REQUEST action', () => {
  expect(clearRequest()).toMatchSnapshot()
})

it('creates a SAVE_REQUEST action', () => {
  expect(saveRequest(123)).toMatchSnapshot()
})
