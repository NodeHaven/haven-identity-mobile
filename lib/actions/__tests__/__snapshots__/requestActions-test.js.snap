// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a AUTHORIZE_REQUEST action 1`] = `
Object {
  "activityId": 123,
  "type": "AUTHORIZE_REQUEST",
}
`;

exports[`creates a CANCEL_REQUEST action 1`] = `
Object {
  "activityId": 123,
  "type": "CANCEL_REQUEST",
}
`;

exports[`creates a CLEAR_REQUEST action 1`] = `
Object {
  "type": "CLEAR_REQUEST",
}
`;

exports[`creates a HANDLE_MESSAGE action with default postback 1`] = `
Object {
  "message": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJpc3MiOiIzNHdqc3h3dmR1YW5vN05GQzh1ak5KbkZqYmFjZ1llV0E4bSIsImlhdCI6MTQ4NTMyMTEzMywiY2xhaW1zIjp7Im5hbWUiOiJCb2IifSwiZXhwIjoxNDg1NDA3NTMzfQ.sg1oJ7J_f2pWaX2JwqzA61oWMUK5v0LYVxUp3PvG7Y25CVYWPyQ6UhA7U9d4w3Ny74k7ryMaUz7En5RSL4pyXg",
  "type": "HANDLE_MESSAGE",
}
`;

exports[`creates a HANDLE_URL action with default postback 1`] = `
Object {
  "clientId": null,
  "popup": true,
  "postback": false,
  "type": "HANDLE_URL",
  "url": "ethereum:0x",
}
`;

exports[`creates a HANDLE_URL action with postback 1`] = `
Object {
  "clientId": null,
  "popup": true,
  "postback": true,
  "type": "HANDLE_URL",
  "url": "ethereum:0x",
}
`;

exports[`creates a HANDLE_URL action with postback, no popup and client id 1`] = `
Object {
  "clientId": "0x010203",
  "popup": false,
  "postback": true,
  "type": "HANDLE_URL",
  "url": "ethereum:0x",
}
`;

exports[`creates a HANDLE_URL action without postback 1`] = `
Object {
  "clientId": null,
  "popup": true,
  "postback": false,
  "type": "HANDLE_URL",
  "url": "ethereum:0x",
}
`;

exports[`creates a SAVE_REQUEST action 1`] = `
Object {
  "activityId": 123,
  "type": "SAVE_REQUEST",
}
`;

exports[`creates a SELECT_REQUEST action 1`] = `
Object {
  "activityId": 123,
  "type": "SELECT_REQUEST",
}
`;

exports[`creates a SELECT_REQUEST_FOR_ACCOUNT action 1`] = `
Object {
  "account": "0x234",
  "activityId": 123,
  "type": "SELECT_ACCOUNT_FOR_REQUEST",
}
`;
